### DISCUSSION TITLE
[Question] Randomization of ground plane friction.

### DISCUSSION THREAD

**Discussion by timswim** [isaac-sim/IsaacLab#1579] [Q&A] [answered]
### Question

Hello! I would like to train an agent (wheeled robot) such that the friction of the ground plane changes at certain time intervals. Am I correct in understanding that the existing function mdp.randomize_rigid_body_material is not suitable for this because the ground plane does not have an id and belongs to all environments at once, so I need to write my own function? Would class EventCfg be suitable for this purpose?

Thanks in advance!

**Comment by KyleM73** [comment, marked as answer]
If you have the friction mode set to "multiply" then changing the wheel friction is equivalent to changing the ground friction, since the ground friction is usually just set to 1.0. It feels weird but physically it's the same, and it makes it easier to use the existing tools like  mdp.randomize_rigid_body_material, just do it for all the wheels at that time interval. I have had trouble with wheels thus far as the primitive cylinder in the gpu pipeline is actually, like, and 18-gon, and even using my own USD with much finer meshes hasn't given very realistic wheel performance, but ymmv. They will be fixing this in an upcoming patch to PhysX/Isaac Sim.

**Reply by timswim** [reply]
That makes sense, thank you very much!

**Comment by kousheekc** [comment]
Just came across this discussion as I want to do something similar (randomise the friction parameters of an object in the scene at intervals) but the docstrings in the function mention the following:     

```
.. attention::
        This function uses CPU tensors to assign the material properties. It is recommended to use this function
        only during the initialization of the environment. Otherwise, it may lead to a significant performance
        overhead.
```

How much of an impact does this actually have?
![screenshot](https://github.com/user-attachments/assets/d370ccb5-c743-475c-a757-9669983e6129)
